jte/src/main/java/gg/jte/resolve/CompositeCodeResolver.java

```diff
@@ -0,0 +1,68 @@
+package gg.jte.resolve;
+
+
+import gg.jte.CodeResolver;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Resolves template code from multiple other resolvers
+ */
+public class CompositeCodeResolver implements CodeResolver {
+
+    private final List<CodeResolver> codeResolvers;
+
+    public CompositeCodeResolver(List<CodeResolver> codeResolvers) {
+        this.codeResolvers = codeResolvers;
+    }
+
+    @Override
+    public String resolve(String name) {
+        var content = "";
+        for (CodeResolver codeResolver : this.codeResolvers) {
+            try {
+                String resolve = codeResolver.resolve(name);
+                if(Objects.nonNull(resolve)){
```

Comment:
`resolve != null` would be fine. `Objects.nonNull()` exists mainly to support Stream filters.